controller/src/main/java/org/jboss/as/controller/management/ManagementInterfaceAddStepHandler.java

```diff
@@ -0,0 +1,97 @@
+import org.jboss.msc.service.ServiceRegistry;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+/**
+ * The base add handler for management interfaces.
+ * @author Emmanuel Hugonnet (c) 2016 Red Hat, inc.
+ */
+public abstract class ManagementInterfaceAddStepHandler extends AbstractAddStepHandler {
+
+    protected ManagementInterfaceAddStepHandler(final RuntimeCapability capability, final AttributeDefinition[] attributeDefinitions) {
+        super(capability, attributeDefinitions);
+    }
+
+   @Override
+    protected boolean requiresRuntime(OperationContext context) {
+        //TODO Gigantic HACK to disable the runtime part of this for the core model testing.
+        //The core model testing currently uses RunningMode.ADMIN_ONLY, but in the real world
+        //the http interface needs to be enabled even when that happens.
+        //I don't want to wire up all the services unless I can avoid it, so for now the tests set this system property
+        return WildFlySecurityManager.getPropertyPrivileged("jboss.as.test.disable.runtime", null) == null;
+    }
+
+    protected void addVerifyInstallationStep(OperationContext context, List<ServiceName> requiredServices) {
+        if(context.isBooting()) {
+            context.addStep(new VerifyInstallationStep(requiredServices), OperationContext.Stage.VERIFY);
+        }
+    }
+
+    protected String asStringIfDefined(OperationContext context, AttributeDefinition attribute, ModelNode model) throws OperationFailedException {
+        ModelNode attributeValue = attribute.resolveModelAttribute(context, model);
+        return attributeValue.isDefined() ? attributeValue.asString() : null;
+    }
+
+    private static class VerifyInstallationStep implements OperationStepHandler {
+        private final List<ServiceName> requiredServices;
+
+        private VerifyInstallationStep(List<ServiceName> requiredServices) {
+            assert requiredServices != null;
+            this.requiredServices = requiredServices;
+        }
+
+        @Override
+        public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+            List<ServiceName> failures = new ArrayList<>();
+            ServiceRegistry registry = context.getServiceRegistry(false);
+            for (ServiceName serviceName : requiredServices) {
+                try {
+                    ServiceController<?> controller = registry.getService(serviceName);
+                    if (controller == null || State.UP != controller.getState()) {
+                        failures.add(serviceName);
+                    }
+                } catch (ServiceNotFoundException ex) {
+                    failures.add(serviceName);
+                }
+            }
+            if (failures.containsAll(requiredServices)) {
+                context.setRollbackOnly();
+                ModelNode requiredServicesNode = context.getFailureDescription().get(ROOT_LOGGER.missingRequiredServices());
```

Comment:
You don't need to deal with the failure description. The existing verification steps will already output the correct  failure description. All this verification needs to do is force the rollback with the context.setRollbackOnly().